### DISCUSSION TITLE
BPS/EnergyPlus, problem with IDF files

### DISCUSSION THREAD

**Discussion by Xingjian-Wang** [BIM2SIM/bim2sim#498] [Q&A] [answered]
Dear team,

A mate and I, who are both working on BIM2SIM related thesis, have encountered a common issue. We are able to generate IDF files using BIM2SIM, but when attempting to run the IDF files in EnergyPlus, various errors occur and we are unable to obtain simulation results. I have attempted to resolve all severe errors, but when running the simulation in EnergyPlus, it crashes, and I am not able to see the further error.

So far, we have tried four building models in IFC format, including the building models of our respective projects (multi-story buildings with an area of over 10,000 square meters), the main building of EBC ERC in the example folder, and AC20-FZK-Haus. Currently, only AC20-FZK-Haus can generate an IDF file without errors. Our IFC files meet the requirements of BIM2SIM for IFC4 files. I would like to know if there are any additional settings required for exporting IFC files or if there are any other potential reasons for these issues.

Thank you in advance for any guidance and suggestions you can provide.

Best Regards,
Xingjian Wang

**Comment by veronikarichter** [comment]
Dear @Xingjian-Wang, 

the IFC4 files must include space boundaries (2nd Level). In most of the authoring tools, these are not exported to IFC by default, so this must be set in the export options. When generating the IDF file, you should have a look at the bim2sim logging, to see if an error occurs somewhere in the IDF generation process. As bim2sim is an open-source project, you are welcome to add further modifications that are required to use your IFC files. 

Have you already checked if the EnergyPlus IDF includes geometry? Otherwise, the geometric export may have failed (or even failed before). I recommend you to check the IDF geometry by importing it to OpenStudio (use OpenStudio as a viewer). Otherwise, if the FZK Haus works fine (and simulates), try to further debug your files by comparing the included objects within the IDF file using the IDF editor.

I hope this helps.

**Comment by Xingjian-Wang** [comment]
Dear Veronika,

many thanks for your answer!
For IFC4 file, yes, they include 2nd level. 
In the log, there is no error about the creat idf, like here:
 [INFO] bim2sim.Playground.run_task: Successfully finished Task '<Task (CreateIdf)>'. 

When I ran the AC20-FZK-Haus file, the task "EnrichMaterial" did not prompt me to input the construction year of the building, whereas for all other ifc files, it required the input of the construction year as shown here: 
[Decision 1/1] Enter value for year_of_construction of <Building> in [year]. 
Additional commands are cancel. Enter value: 2014.
I would like to know if there are any pre-settings for the AC20-FZK-Haus file. Additionally, I have discovered that when the input construction year is later than 2016, BIM2SIM will not be able to generate an IDF file.

Regarding the geometry of the IDF, I have previously attempted to open the generated IDF files in SketchUp. Although these IDFs could not successfully simulate energy results in EnergyPlus, their geometry could still be displayed in SketchUp. Unfortunately, my SketchUp trial version has just expired, and I am unable to show screenshots. In OpenStudio, complete geometry cannot be displayed, and some IDFs display only fragments while others cannot show any geometry at all. I would like to know why the geometry cannot be generated and how it can be adjusted.
![image](https://user-images.githubusercontent.com/72249299/228655990-edad3c48-aec9-4a04-993b-a76a60db7b50.png)


 I have also tried to simplify the exported IFC file and found that IDF files without errors can be generated when there are fewer thermal zones. I have attempted to delete the names of most rooms in the Revit model, leaving only a few named rooms, so that BIM2SIM would only recognize a few thermal zones. IDF files generated this way can simulate energy results in EnergyPlus. I would like to know if additional settings are required for more complex buildings. For example, the EBC main building model in the example file cannot generate an IDF file that can successfully simulate energy in EnergyPlus. Compared to AC20-FZK-Haus, the EBC main building is much more complex.

Many thanks in advance

Best Regards,
Xingjian Wang

**Reply by Xingjian-Wang** [reply]
Hi Veronika,

I have an further question. I tried to set the "LOD" to "medium" and choose to bind thermal zones by usage in the decision section to create fewer thermal zones in the IDF file and avoid the possibility of errors. 
![image](https://user-images.githubusercontent.com/72249299/228975107-e31d8d83-3f7f-44e6-975a-00e354271d86.png)

Unfortunately, BIM2SIM was unable to generate the IDF file. Do you know any solutions to this problem? The screenshot below shows the error generated when creating the IDF.
![image](https://user-images.githubusercontent.com/72249299/228975135-118e1b6b-6ba3-456d-95cd-52b2f7bc1933.png)

Thanks in advance

Best Regards,
Xingjian Wang

**Reply by veronikarichter** [reply, marked as answer]
> Dear Veronika,
> 
> many thanks for your answer! For IFC4 file, yes, they include 2nd level. In the log, there is no error about the creat idf, like here: [INFO] bim2sim.Playground.run_task: Successfully finished Task '<Task (CreateIdf)>'.
> 
> When I ran the AC20-FZK-Haus file, the task "EnrichMaterial" did not prompt me to input the construction year of the building, whereas for all other ifc files, it required the input of the construction year as shown here: [Decision 1/1] Enter value for year_of_construction of in [year]. Additional commands are cancel. Enter value: 2014. I would like to know if there are any pre-settings for the AC20-FZK-Haus file. Additionally, I have discovered that when the input construction year is later than 2016, BIM2SIM will not be able to generate an IDF file.
> 
> Regarding the geometry of the IDF, I have previously attempted to open the generated IDF files in SketchUp. Although these IDFs could not successfully simulate energy results in EnergyPlus, their geometry could still be displayed in SketchUp. Unfortunately, my SketchUp trial version has just expired, and I am unable to show screenshots. In OpenStudio, complete geometry cannot be displayed, and some IDFs display only fragments while others cannot show any geometry at all. I would like to know why the geometry cannot be generated and how it can be adjusted. ![image](https://user-images.githubusercontent.com/72249299/228655990-edad3c48-aec9-4a04-993b-a76a60db7b50.png)
> 
> I have also tried to simplify the exported IFC file and found that IDF files without errors can be generated when there are fewer thermal zones. I have attempted to delete the names of most rooms in the Revit model, leaving only a few named rooms, so that BIM2SIM would only recognize a few thermal zones. IDF files generated this way can simulate energy results in EnergyPlus. I would like to know if additional settings are required for more complex buildings. For example, the EBC main building model in the example file cannot generate an IDF file that can successfully simulate energy in EnergyPlus. Compared to AC20-FZK-Haus, the EBC main building is much more complex.
> 
> Many thanks in advance
> 
> Best Regards, Xingjian Wang

Dear Xingjian, 

during developing the bim2sim code, we used the FZK Haus for testing. Also, the FZK Haus IFC is an IFC file in "research quality", so most of the parameters that an IFC could provide are included here. The material is enriched automatically here (or maybe directly taken from IFC materials). This material enrichment may need more user input for your models.

Construction year: currently, our enrichment work with templates up to the year 2015 (we don't have template data for more recent years). You need to check anyway if our template data is applicable for your cases. Its likely that you should update your materials in the IDF to get an IDF that matches the actual setup of your building. It is your responsibility to test and modify that after using our tool.

We had some troubles recently for models that use millimeters as length unit (see issue #490). Feel free to contribute here to fix this issue in a new pull request. If you model your building using meters instead of millimeters, this might solve the display problem. But maybe your problem is somewhere else. Main part of setting up an energy model is to figure out where the bugs are and solve them. 

In general, our tool is not limited to a certain amount of zones. The success of our tool though highly depends on the quality of the input IFC model. FZK Haus works well since the quality of the IFC standard is quite high (as it has been designed for research purposes). Errors in your model may result from the unit conversion bug (millimeters) but also may result from other issues regarding differences in modeling standards. One known issue is that bim2sim currently cannot handle curtain walls (see #223, these are just ignored and should not cause any errors but gaps in the geometry). 

I hope this helps. If error reports within IDF do not help you, I'd recommend you to go through the text file manually and look for potential errors. Sometimes also updating the IDF to a more recent version helps. For further help how to debug EnergyPlus I'd recommend to go through the EnergyPlus documentation ([https://energyplus.net/](https://energyplus.net/)) and [https://unmethours.com/](https://unmethours.com/questions/).
